service/src/test/java/uk/nhs/adaptors/oneoneone/cda/report/mapper/ConsentMapperTest.java

```diff
@@ -0,0 +1,157 @@
+    private POCDMT000002UK01ClinicalDocument1 clinicalDocument;
+
+    @Mock
+    private Encounter encounter;
+
+    @Mock
+    private Period period;
+
+    @Mock
+    private POCDMT000002UK01Section section;
+
+    @Mock
+    private POCDMT000002UK01Component5 component5;
+
+    @Mock
+    private ST title;
+
+    @Mock
+    private CE code;
+
+    @Mock
+    private CS cs;
+
+    @Mock
+    private II ii;
+
+    @Mock
+    private StrucDocText strucDocText;
+
+    @Mock
+    private POCDMT000002UK01Consent authConsent;
+
+    @Mock
+    private POCDMT000002UK01Entry entry;
+
+    @InjectMocks
+    private ConsentMapper consentMapper;
+
+    @Before
+    public void setUp() {
+        POCDMT000002UK01Component2 component2 = mock(POCDMT000002UK01Component2.class);
+        POCDMT000002UK01Component3 component3 = mock(POCDMT000002UK01Component3.class);
+        POCDMT000002UK01Authorization authorization = mock(POCDMT000002UK01Authorization.class);
+        POCDMT000002UK01StructuredBody structuredBody = mock(POCDMT000002UK01StructuredBody.class);
+
+        code = CE.Factory.newInstance();
+        code.setCodeSystem(ITK_SNOMED);
+        code.setCode(INFORMATION_ADVICE_GIVEN);
+
+        title.setLanguage(LANG);
+        cs.setCode(LANG);
+
+        when(clinicalDocument.getComponent()).thenReturn(component2);
+        when(clinicalDocument.sizeOfAuthorizationArray()).thenReturn(1);
+        when(clinicalDocument.getAuthorizationArray()).thenReturn(new POCDMT000002UK01Authorization[]{authorization});
+        when(clinicalDocument.getSetId()).thenReturn(ii);
+        when(ii.getRoot()).thenReturn(ROOT_ID);
+
+        when(authorization.getConsent()).thenReturn(authConsent);
+        when(authConsent.getCode()).thenReturn(code);
+
+        when(component2.getStructuredBody()).thenReturn(structuredBody);
+        when(component3.getSection()).thenReturn(section);
+        when(component5.getSection()).thenReturn(section);
+        when(structuredBody.getComponentArray()).thenReturn(new POCDMT000002UK01Component3[]{component3});
+    }
+
+    @Test
+    public void shouldMapITKReportToConsent() {
+        Patient patient = mock(Patient.class);
+        Reference patientReference = new Reference(patient);
+
+        mockEncounter(patient, patientReference);
+        mockSection();
+
+        Consent consent = consentMapper.mapConsent(clinicalDocument, encounter);
+
+        assertThat(consent).isNotNull();
+        assertThat(consent.getIdElement().getValue()).startsWith(URN_UUID);
+        assertThat(consent.getStatus()).isEqualTo(Consent.ConsentState.ACTIVE);
+
+        assertThat(consent.hasDateTime()).isTrue();
+        assertThat(consent.getLanguage()).isEqualTo(LANG);
+        assertThat(consent.getPeriod()).isEqualTo(period);
+        assertThat(consent.getText()).isNotNull();
+        assertThat(consent.getPolicyRule()).isEqualTo(OPT_OUT_URI);
+        CodeableConcept action = consent.getActionFirstRep();
+        if (!action.isEmpty()) {
```

Comment:
aren't you able to predict if action is going to be empty?

Reply:
Fixed as this was failing as display name was missing. Now added